src/main/java/duke/ChatBot.java

```diff
@@ -0,0 +1,170 @@
+package duke;
+
+import java.time.LocalDateTime;
+import java.util.ArrayList;
+
+import duke.storage.Storage;
+import duke.tasks.Task;
+import duke.tasks.TaskList;
+
+/**
+ * ChatBot is a class that encapsulates what a chat bot has.
+ * It functions as a chat bot that responds to user inputs and
+ * does tasks specified by the user.
+ */
+public class ChatBot {
+
+    private ArrayList<Task> temp;
+    private TaskList tasklist;
+    private Storage s;
+    private UI ui;
+
+    private int exitStatus = 1;
+
+    /**
+     * This is the constructor of ChatBot that initializes the UI and Storage.
+     */
+    public ChatBot() {
+        this.s = new Storage();
+        this.ui = new UI();
+    }
+
+    /**
+     * This is the method called when the program is started.
+     * It checks for the save directory and loads any previously
+     * stored tasks. It also prints out the start message.
+     */
+    public void start() {
+        s.makeDir();
+        s.checkFile();
+        this.temp = s.loadTasks();
+        tasklist = new TaskList(temp);
+    }
+
+    /**
+     * This method is to stop the program from running and prints
+     * out the exit message.
+     */
+    public String handleExit() {
+        exitStatus = 0;
+        return ui.showExitMessage();
+    }
+
+    /**
+     * This method is to call the UI to print all the tasks that is stored.
+     */
+    public String handleList() {
+        return ui.getListMessage(tasklist);
+    }
+
+    /**
+     * This method is to mark the user specified task as done
+     * and print out the corresponding UI message.
+     *
+     * @param index Index of the task.
+     */
+    public String handleDone(int index) {
+        Task t = tasklist.completeTask(index);
+        s.saveTasks(tasklist);
+        return ui.showCompleteTaskMessage(t);
+    }
+
+    /**
+     * This method is to add a new deadline to the storage and
+     * print out the add message when the task is added successfully.
```

Comment:
Personally I like this header, its very clear and neat! You could also consider just beginning the description with 'returns' or 'adds' or 'sends' etc. So for instance, 'adds a new deadline ....'.

Reply:
Will update the javadocs to fit documentation standards